### PR TITLE
Create component tree using model method.

### PR DESCRIPTION
**Pull request by rhyw**
Part of fix for OLCS-447

### CODE REVIEW

**Comment by rhyw** [comment]
The update of "ProductTreeNode" is not included in the MR, but should be similar.

**Comment by rhyw** [comment]
Some of the changes e.g., `update_or_create_component` is used in https://github.com/RedHatProductSecurity/openlcs/pull/136 . You may need to resolve conflicts.

**Comment by RichardXuan** [comment]
Updated

**Comment by rhyw** [comment]
@RichardXuan btw, one pull could have multiple commits. You could commit your own changes and push to this PR, so that it's easier to track newly added changes from log.

**Comment by RichardXuan** [comment]
@rhyw Ok, got it

File: openlcs/packages/models.py [hunk 1]
```diff
@@ -252,8 +252,15 @@ def update_or_create_component(component_data):
...
             'purl': component_data.get('purl', ''),
+
+            "summary_license": "" if component_data.get('summary_license') \
+            is None else component_data['summary_license']
         }
-        component, _ = Component.objects.update_or_create(
+
+        if "is_source" in component_data:
+            defaults['is_source'] = component_data["is_source"]
+
```
Comment (rhyw) [inline openlcs/packages/models.py:262]: Thanks for the additional setup here.

File: openlcs/products/models.py [hunk 2]
```diff
@@ -178,6 +203,29 @@ class Meta:
             )
         ]
 
+    @classmethod
+    def build_release_node(cls, release_component: dict, product_release: str):
```
Comment (rhyw) [inline openlcs/products/models.py:207]: And here!

**Comment by rhyw** [comment]
LGTM

**Pull request merged**